dbt/adapters/snowflake/impl.py

```diff
@@ -254,3 +260,33 @@ def submit_python_job(self, parsed_model: dict, compiled_code: str):
 
     def valid_incremental_strategies(self):
         return ["append", "merge", "delete+insert"]
+
+    def drop_relation(self, relation):
+        with RelationDBSubstitution(relation):
```

Comment:
why is this implemented with a context manager that mutates the relation, and not just do `copy(relation)` and replace the relevant field? so it wont need to deal with cleanup at all?

Reply:
It at least needs to be a deep copy since I am substituting second level field (relation.path.database). So who knows which complexities this will cause. My solution is simple and works. Do you expect issues with it? Multi-tasking can be an issue in theory, but in practice the class uses a single snowflake adapter, which is not thread safe.